src/zdns/conf.go

```diff
@@ -49,8 +49,17 @@ const (
 	StatusIterTimeout  Status = "ITERATIVE_TIMEOUT"
 	StatusNoAuth       Status = "NOAUTH"
 	StatusNoNeededGlue Status = "NONEEDEDGLUE" // When a nameserver is authoritative for itself and the parent nameserver doesn't provide the glue to look it up
+	StatusCircular     Status = "CIRCULAR"     // When circular query dependencies are detected
 )
 
+func isStatusRetryable(status Status) bool {
+	switch status {
+	case StatusServFail, StatusNXDomain, StatusRefused, StatusTruncated, StatusError, StatusTimeout, StatusIterTimeout:
```

Comment:
@phillip-stephens could you check if these make sense to you

Reply:
Since `retries` are global to the lookup, we shouldn't use a `retry` if there's no chance of getting back a different response.

`NXDomain` we should not retry. Based on [RFC 8020](https://datatracker.ietf.org/doc/html/rfc8020), there's no chance we re-query and get a different response to an NXDomain. So this **should not** be retry-able
The others I'm fine with leaving as-is.

Additionally, let's add `NXDomain` as a specific `case` statement and add that RFC as reason why we will not be re-trying that specific domain.

And make `StatusTimeout` and `StatusIterTimeout` their own case statement and comment:
`up to caller's implementation to retry` since technically where this function is used in lookup.go, we'll abort the query if the `ctx` expires, the main reason for getting this error.

The others I'm fine with!